paasta_tools/cli/cmds/check.py

```diff
@@ -202,44 +203,34 @@ def yaml_check(service_path):
         paasta_print(PaastaCheckMessages.YAML_MISSING)
 
 
-def get_chronos_steps(service, soa_dir):
-    """This is a kind of funny function that gets all the chronos instances
-    for a service and massages it into a form that matches up with what
+def get_framework_steps(framework, service, soa_dir):
+    """This is a kind of funny function that gets all the instances for specified
+    service and framework, and massages it into a form that matches up with what
     deploy.yaml's steps look like. This is only so we can compare it 1-1
-    with what deploy.yaml has for linting."""
-    steps = []
-    for cluster in list_clusters(service, soa_dir):
-        for _, instance in get_service_instance_list(
-                service=service,
-                cluster=cluster,
-                instance_type='chronos',
-                soa_dir=soa_dir,
-        ):
-            config = load_chronos_job_config(
-                service=service,
-                instance=instance,
-                cluster=cluster,
-                soa_dir=soa_dir,
-                load_deployments=False,
-            )
-            steps.append(config.get_deploy_group())
-    return steps
+    with what deploy.yaml has for linting.
 
+    :param framework: one of 'marathon', 'chronos', 'adhoc'
+    :param service: the service name
+    :param soa_dir: The SOA configuration directory to read from
+
+    :returns: a list of deploy group names used by the service.
+    """
+
+    load_config_func = {
+        'marathon': load_marathon_service_config,
+        'chronos': load_chronos_job_config,
+        'adhoc': load_adhoc_job_config,
+    }
 
-def get_marathon_steps(service, soa_dir):
-    """This is a kind of funny function that gets all the marathon instances
-    for a service and massages it into a form that matches up with what
-    deploy.yaml's steps look like. This is only so we can compare it 1-1
-    with what deploy.yaml has for linting."""
     steps = []
     for cluster in list_clusters(service, soa_dir):
         for _, instance in get_service_instance_list(
                 service=service,
                 cluster=cluster,
-                instance_type='marathon',
+                instance_type=framework,
                 soa_dir=soa_dir,
         ):
-            config = load_marathon_service_config(
+            config = load_config_func[framework](
```

Comment:
It makes a lot of sense to not duplicate functionality in the codebase and to reduce the scope of this function though.

How about this as a compromise: make `get_instance_config` take an optional `instance_type` arg, and if none it will look it up. And here we pass it in.

I'm trying to reduce the future effort of adding new instance_types, having the logic about how to read them in in only one function will help.